secret-rotator/svckey/svckey.go

```diff
@@ -92,10 +92,11 @@ func (p *Provisioner) Deactivate(labels map[string]string, version string) error
 	// the reason for the prefix "v" is that Secret Manager labels need to begin with a lowwer case letter
 	name := fmt.Sprintf("projects/%s/serviceAccounts/%s@%s.iam.gserviceaccount.com/keys/%s", labels["project"], labels["service-account"], labels["project"], labels["v"+version])
 
-	_, err := p.Service.Projects.ServiceAccounts.Keys.Delete(name).Do()
-	if err != nil {
-		return fmt.Errorf("Projects.ServiceAccounts.Keys.Delete: %v", err)
-	}
+	// temporarily disabling service account key deletion, for safety reasons.
+	// _, err := p.Service.Projects.ServiceAccounts.Keys.Delete(name).Do()
+	// if err != nil {
+	// 	return fmt.Errorf("Projects.ServiceAccounts.Keys.Delete: %v", err)
+	// }
```

Comment:
Can you gate this behind a flag? e.g. plumb through NewProvisioner?

Reply:
I added a flag `--enable-deletion`  to controller this, which is default to false for now.